src/coreclr/jit/decomposelongs.cpp

```diff
@@ -1507,6 +1511,50 @@ GenTree* DecomposeLongs::DecomposeRotate(LIR::Use& use)
     }
 }
 
+//------------------------------------------------------------------------
+// DecomposeSelect: Decompose 64-bit GT_SELECT into a 32-bit GT_SELECT and
+// 32-bit GT_SELECT_HI.
+//
+// Arguments:
+//    use - the LIR::Use object for the def that needs to be decomposed.
+//
+// Return Value:
+//    The next node to process.
+//
+GenTree* DecomposeLongs::DecomposeSelect(LIR::Use& use)
```

Comment:
Am I right in thinking this will only happen on X64 (and not Arm64) ?

Reply:
Decomposition is only for 32-bit targets, so only on x86 (and arm32 if it was supported).